File: microservices.springframework/api-gateway-service/src/main/java/swd392/apigatewayservice/dto/ApiResponse.java
package swd392.apigatewayservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;

/*
 * File: ApiResponse.java
 * Description: Response object for API calls.
 *
 * Version History:
 * ----------------------------------------------------------------------------
 * v1.0 - ?/2/2025 - Nguyen Tien Thuan - Define class.
 * ----------------------------------------------------------------------------
 *
 * Author(s): Nguyen Tien Thuan
 * Last Modified: 26/3/2025
 * Notes:
 */
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ApiResponse<T> {
    @JsonProperty("status")
    public String status;

    @JsonProperty("message")
    public String message;

    @JsonProperty("dataResponse")
    public T dataResponse;
}

File: microservices.springframework/cloud-storage-service/src/main/kotlin/aas/cloudstorageservice/application/dto/ApiResponse.kt
package aas.cloudstorageservice.application.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class ApiResponse<Any> (
    @JsonProperty("status")
    var status: String? = null,

    @JsonProperty("message")
    var message: String? = null,

    @JsonProperty("dataResponse")
    var dataResponse: Any? = null
)
